modyn/supervisor/internal/grpc_handler.py

```diff
@@ -654,11 +599,6 @@ def wait_for_evaluation_completion(
             working_queue.append(current_evaluation_id)
             sleep(1)
 
-        self.pipeline_status_queue.put(
-            pipeline_stage_msg(
-                PipelineStage.EVALUATION_COMPLETED, MsgType.ID, id_submsg(IdType.TRAINING, training_id), True
-            )
-        )
```

Comment:
I think most of changes in this file are unrelated with the purpose of this PR. And `if key not in dict` does not look worse than `if dict.get(key)`. To make sure we didn't accidentally introduce something wrong in the request preparation, I suggest we keep the old code.

Reply:
In general I advocate the principle "if there is not a really justified reason for a change, then keep it the same".

A pure code style change just adds more work to the reviewer and more risk to uncaught bugs (for example [here](https://github.com/eth-easl/modyn/pull/435/files#r1606098572), hard to catch by the reviewer), because our unit tests certainly do not cover a lot of cases. (Just passing all the tests does not mean it is risk-free). And the old code has less risk to go wrong because we have used it for long.

I hope my thoughts make sense!

Reply:
Ah now I see your point in https://github.com/eth-easl/modyn/pull/435/files#r1603030383. Yes in that way this makes sense. Sorry for the misunderstanding!